### PR TITLE
Constructor with params in CdiContainer

### PR DESCRIPTION
**Pull request by clairton**

### CODE REVIEW

File: src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java [hunk 1]
```diff
@@ -13,12 +13,32 @@
...
 
-        public void start() {
-                weld = new Weld();
-                weldContainer = weld.initialize();
-                contexts = getContexts();
-        }
+    	public CdiContainer(Weld weld, WeldContainer weldContainer) {
+    		super();
+    		this.weld = weld;
+    		this.weldContainer = weldContainer;
```
Comment (Turini) [inline src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java]: i'm just unsure about allowing user to provide a non-related instance of weld and its container. 
is there a case of use in which the weldcontainer isn't the weld#initialize output?

File: src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java [hunk 2]
```diff
@@ -13,12 +13,32 @@
...
+    		this.contexts = contexts;
+    	}
 
-        public void start() {
-                weld = new Weld();
-                weldContainer = weld.initialize();
-                contexts = getContexts();
-        }
+    	public CdiContainer(Weld weld, WeldContainer weldContainer) {
+    		super();
```
Comment (Turini) [inline src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java]: cut it out ;) in this case, the less, the better

**Comment by Turini** [comment]
this change sounds great, @clairton! I just left a couple comments

File: src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java (hunk 1 above)
Comment (clairton) [inline src/main/java/br/com/caelum/vraptor/test/container/CdiContainer.java]: I have tests with apache deltaspike, before this constructor, the container is started twice.

**Comment by clairton** [comment]
@Turini  everething is ok?

**Comment by Turini** [comment]
awesome. thanks @clairton

**Pull request merged**